core/json/Program.cs

```diff
@@ -146,7 +148,7 @@ public static async Task<(int count, int total)> ReadJsonFromStreamUsingSpan(Str
             return (count, total);
         }
 
-        public static (JsonReaderState state, int count, int total) PartialCountUniversityOf(ReadOnlySpan<byte> dataUtf8, bool isFinalBlock, ref bool foundName, JsonReaderState state)
+        public static (long bytesConsumed, int count, int total) PartialCountUniversityOf(ReadOnlySpan<byte> dataUtf8, bool isFinalBlock, ref bool foundName, ref JsonReaderState state)
```

Comment:
I don't fully understand how this fixes the issue. The code looks identical to me.
I will take a closer look later, but if someone already knows, please share.

Reply:
Ah yes, I forgot we removed `BytesConsumed` from the state and only kept the one on the reader.
https://github.com/dotnet/corefx/pull/38928

https://docs.microsoft.com/en-us/dotnet/api/system.text.json.utf8jsonreader.bytesconsumed?view=netcore-3.1

Also, the pass-by-ref change was a good idea.